staging/src/k8s.io/apimachinery/pkg/api/validation/generic.go

```diff
@@ -70,7 +70,7 @@ var ValidateServiceAccountName = NameIsDNSSubdomain
 // maskTrailingDash replaces the final character of a string with a subdomain safe
 // value if is a dash.
 func maskTrailingDash(name string) string {
-	if strings.HasSuffix(name, "-") {
+	if len(name) > 1 && strings.HasSuffix(name, "-") {
```

Comment:
This looks reasonable, given how it is used.

Please update the godoc to indicate the change in logic, and add unit tests to cover the following cases:
* `""`
* `"-"`
* `"-foo"`
* `"-foo-"`

Reply:
Hey @liggitt, I just had a quick question: to mask the trailing dash, wouldn't `name[:len(name)-1]` suffice instead of the `-2`?

Reply:
We don't want to mask the `-` if it is also the leading character

The point of masking the suffix is that for generated names, a random alpha-numeric is appended, which would make `foo-` a valid dns name. We don't want to replace `-` with `a` if it is also the prefix, since `-`+suffix would not be a valid dns name.

Reply:
Oh I get it now, thank you for the clarification 😄